### ISSUE TITLE
Groupdate won't work with config.active_record.default_timezone = :local

### ISSUE THREAD

**Issue by ranjithnalimela** [ankane/groupdate#66]
Hi 
my aplication.rb is setup like 
 
   #config.active_record.default_timezone = :utc    
    config.time_zone = 'Central Time (US & Canada)'
    config.active_record.default_timezone = :local

when i am started my server its shows following in error,

ActiveRecord::Base.default_timezone must be :utc to use Groupdate

How can user group by date with :local time zone settings

**Comment by bodrovis** [comment]
@ankane I am sorry, but haven't found a clear answer anywhere. Does this mean that it won't work even if only `config.time_zone` is provided? I have this setting redefined in my application.rb, but `config.active_record.default_timezone` is not set. Still, I always get zeros for all dates. Commenting out `config.time_zone` solves the problem. However, this setting is required in my case. Is there anything that can be done?

**Comment by lluis** [comment]
Hi! Is this solved or there's any workaround?

**Comment by ankane** [comment]
Hey @lluis, are you referring to the issue in the original ticket or @bodrovis's issue?

**Comment by lluis** [comment]
@ankane sorry, to the issue in the original ticket, default_timezone set to :local

**Comment by ankane** [comment]
No, Groupdate still requires ActiveRecord::Base.default_timezone to be `utc`.

**Comment by lluis** [comment]
ok, thanks!

**Comment by rocLv** [comment]
@ankane Could you explain a little? Why the default_timezone has to be :utc?

**Comment by ankane** [comment]
Hey @rocLv, Groupdate's queries assume data is stored in UTC, so you'll get the wrong result if it's not. This is why we have it throw an error.

**Comment by rocLv** [comment]
@ankane Could we just ignore 'TIMEZONE'?  It seems 
```sql
select created_at, EXTRACT(HOUR from created_at::timestamptz)::integer as recorde_hour
from wares
```
and 

```sql
select created_at, EXTRACT(HOUR from created_at::timestamptz AT TIME ZONE 'Asia/Shanghai')::integer as recorde_hour
from wares
```

return the same result.

**Comment by mirelon** [comment]
Why is this issue closed? There are other gems which require local time zone (date_time_picker). Does that mean I cannot use this gem now?

**Comment by ankane** [comment]
@mirelon It's closed because there are no plans to add this. You'll have to judge for yourself whether or not you can use it. I'm not familiar with the other gems you're using.

**Comment by rizal76** [comment]
My workaround

In model:
```
 def self.default_timezone
      :utc
 end
```

Or (between groupdate code)
```
// set utc
YourModel.default_timezone = :utc
// your groupdate logic (example)
count_group = YourModel.group_by_period(:day, :created_at).count
// set to your local
YourModel.default_timezone = :utc
```

In your config application
```
    # Since Groupdate use UTC and convert to local ( by default ),
    # so if we use local timezone in db, we don't need to convert to our local timezone anymore,
    # thats why we use UTC
    Groupdate.time_zone = "UTC"
```

**Comment by wtom** [comment]
I've created a monkey patch for my use case. Maybe it helps someone. Please note, i just changed the mysql_adapter, so it will just work with a mysql database.

https://gist.github.com/wtom/875b8fd30b9ad8401d620f774b5f7b40

**Comment by ohsun0405** [comment]
Just an idea. Not sure this works, but it looks good to me.

```
class ApplicationRecord < ActiveRecord::Base
  def self.groupdate(method, column, **options)
    ActiveRecord.default_timezone = :utc
    object = public_send(method, column, **options)
    ActiveRecord.default_timezone = :local
    object
  end
end
```

```
Books.groupdate(:group_by_month, :date, format: "%Y/%m").count
```

**Comment by SimBed** [comment]
I have implemented @rizal76 's very useful workaround for this issue, which works well. The minor niggle I have is that the error reporting software I use, honeybadger sometimes reports the error:

`Groupdate::Error: ActiveRecord.default_timezone must be :utc to use Groupdate`

Before implementing any groupdate logic, I set `MyModel.default_timezone = :utc` as shown in the workaround, so I can't see how this error can occur. Users don't seem to experience any issues. Any ideas what's happening?
